local/include/core/ProxyVerifier.h

```diff
@@ -80,6 +80,25 @@ static const std::string YAML_RULE_CONTAINS{"contains"};
 static const std::string YAML_RULE_PREFIX{"prefix"};
 static const std::string YAML_RULE_SUFFIX{"suffix"};
 
+enum YAML_URL {Scheme, Host, Port, Authority, Path, Query, Fragment, Error};
+
+static const std::string YAML_URL_SCHEME{"scheme"};
+static const std::string YAML_URL_HOST{"host"};
+static const std::string YAML_URL_PORT{"port"};
+static const std::string YAML_URL_AUTHORITY{"authority"};
+static const std::string YAML_URL_PATH{"path"};
+static const std::string YAML_URL_QUERY{"query"};
+static const std::string YAML_URL_FRAGMENT{"fragment"};
+
+static const std::size_t URL_PART_NUM = 7;
+
+static const swoc::TextView PART_TO_NAME[URL_PART_NUM] = {
```

Comment:
We might look at `swoc::Lexicon` for this.

Reply:
It's not because you can use static_cast<size_t> on a YAML_URL to get an index into the array. So no need to do hashing. It's still effectively a type of "map", but with the keys as consecutive size_ts.